lightning/src/onion_message/messenger.rs

```diff
@@ -1426,7 +1464,10 @@ where
 			message_recipients
 				.iter()
 				.filter(|(_, peer)| matches!(peer, OnionMessageRecipient::ConnectedPeer(_)))
-				.map(|(node_id, _)| *node_id)
+				.map(|(node_id, _)| MessageForwardNode {
+					node_id: *node_id,
+					short_channel_id: None,
+				})
```

Comment:
@TheBlueMatt This means even if we are configured with a `DefaultMessageRouter`, one of the hops in the created reply path won't be compact. Not sure if there is much we could do about this, but I just want to call it out. This affects three uses:
- `OnionMessenger::send_onion_message` for stand-alone onion message sending
- `OnionMessenger::handle_onion_message_response` or async response handling
- `OnionMessenger::next_onion_message_for_peer` for sync response handling

Prior to this change we'd always use non-compact paths when creating reply paths. But now it is based on the configured `MessageRouter` except here where we don't have an SCID for the peer.

Reply:
I suppose we could parameterize `OnionMessenger` with a `NetworkGraph` instead of a `MessageRouter` such that we can create a `NodeIdMessageRouter` internally? This would at least make those reply path hops consistent. However, when using `ChannelMangaer::pay_for_offer` we'll still make reply paths using (exclusively) compact hops.